File: OctopusNode/src/main/java/com/octopus/transport/SubJob.java
package com.octopus.transport;

import com.octopus.node.core.JobType;

public class SubJob {
    private final int id;
    private final String parentUUID;
    private final Object data;
    private final int sequenceNumber;
    private final JobType subJobType;

    public SubJob(int id, String parentUUID, Object data, int sequenceNumber, JobType subJobType) {
        this.id = id;
        this.parentUUID = parentUUID;
        this.data = data;
        this.sequenceNumber = sequenceNumber;
        this.subJobType = subJobType;
    }

    public int getId() {
        return id;
    }

    public String getParentUUID() {
        return parentUUID;
    }

    public Object getData() {
        return data;
    }

    public int getSequenceNumber() {
        return sequenceNumber;
    }

    public JobType getSubJobType() {
        return subJobType;
    }
}

File: OctopusService/src/main/kotlin/com/octopus/core/models/SubJob.kt
package com.octopus.core.models

import com.octopus.core.enums.JobType
import java.util.*

class SubJob(
        val parentUUID: String,
        var data: Any,
        var sequenceNumber: Int,
        var subJobType: JobType
) {
    var id: Int = 0
    var uuid: String = UUID.randomUUID().toString()
    var retries: Int = 0

    constructor(id: Int, uuid: String, parentUUID: String, data: Any, sequenceNumber: Int, subJobType: JobType, retries: Int):
            this(parentUUID, data, sequenceNumber, subJobType) {
        this.id = id
        this.uuid = uuid
        this.retries = retries
    }

    override fun toString(): String {
        return "SubJob(parentUUID='$parentUUID', data=$data, sequenceNumber=$sequenceNumber, subJobType=$subJobType, id=$id, uuid='$uuid')"
    }

    fun withData(data: Any) : SubJob {
        this.data = data
        return this
    }
}